case-lib/lib.sh

```diff
@@ -151,6 +151,39 @@ find_ldc_file()
     printf '%s' "$ldcFile"
 }
 
+ZPH_LOG_COLLECT=0
+func_cavstool_log_collect()
+{
+    local logopt="-l"
+
+    if [ -z "$CAVSTOOL" ]; then
+        CAVSTOOL=$(command -v cavstool.py) || {
+            dlogw 'No cavstool.py found in PATH'
+            return 1
+        }
+    fi
+
+    test -x "$CAVSTOOL" || {
+        dlogw "$CAVSTOOL not found or not executable"
+        return 2
+    }
+
+    logfile=$LOG_ROOT/cavstool.txt
+
+    if func_hijack_setup_sudo_level ;then
+        # shellcheck disable=SC2034 # external script will use it
+        ZPH_LOG_COLLECT=1
+    else
+        >&2 dlogw "without sudo permission to run $CAVSTOOL command"
+        return 3
```

Comment:
You're returning 3 different error codes but the callers never look at them. I don't think you ever tested all this error handling code; what problem does it solve? sudo, the shell and other system utilities already provide error messages for free, what was wrong with them?

Every line of code you submit should have been run at least once. Write the code you actually need, not the code you think you need.